[sessions] When using an active session with cy.session waiting for an array of aliased XHR's do not resolve

### Current behavior

When using cy.session to create a session and login using this session, it appears that XHR requests defined and waited on seem to hang, with no error in the Cypress test runner. This seems to happen in an array of aliases to wait

`cy.wait('['@alias1',  '@alias2', '@alias3']`

### Desired behavior

Cypress should wait for all aliases defined in the array.

### Test code to reproduce

```js
const login = (name, password) => {
  cy.session(
    [name, password],
    () => {
      cy.visit('/')
      cy.get('#signin-username').type(name)
      cy.get('#signin-password').type(password)
      cy.get('#signin-submit').click()
    })
}
```

```js   
login(Cypress.env('testUserEmail'), Cypress.env('testUserPassword'))
    cy.visit('/')
    cy.wait(['@alias1', '@alias2', '@alias3']);
```

### Cypress Version

8.2.0

### Other

_No response_

This happened to me when running headless

```javascript
Cypress.Commands.add('loginAs', (mail, password) => {
  cy.session('defaultUser', () => {
    cy.visit(Cypress.env('base_url'));
    cy.get('[name="email"]')
      .type(mail)
      .get('[name="password"]')
      .type(password)
      .get('[data-cy=login-button]')
      .click()
      .url().should('contain', '/oauth/consent');   
  });
});
```
After logging it should wait for the URL assertion but it gets stuck in the login form. This problem does not happen in headed Cypress.

Error:
```
  1) login:
     AssertionError: Timed out retrying after 4000ms: expected 'http://localhost:8001/login' to include '/oauth/consent'
      at Object.eval [as setup] (http://abowire.localhost:8001/__cypress/tests?p=cypress/support/index.js:131:14)

--- a/packages/server/lib/project-base.ts
+++ b/packages/server/lib/project-base.ts
@@ -39,7 +39,7 @@ import { checkSupportFile } from './project_utils'
 // TODO: Figure out how to type this better.
 type ReceivedCypressOptions =
   Partial<Pick<Cypress.RuntimeConfigOptions, 'hosts' | 'projectName' | 'clientRoute' | 'devServerPublicPathRoute' | 'namespace' | 'report' | 'socketIoCookie' | 'configFile' | 'isTextTerminal' | 'isNewProject' | 'proxyUrl' | 'browsers'>>
-  & Partial<Pick<Cypress.ResolvedConfigOptions, 'supportFolder' | 'experimentalSourceRewriting' | 'fixturesFolder' | 'reporter' | 'reporterOptions' | 'screenshotsFolder' | 'pluginsFile' | 'supportFile' | 'integrationFolder' | 'baseUrl' | 'viewportHeight' | 'viewportWidth' | 'port' | 'experimentalInteractiveRunEvents'>>
+  & Partial<Pick<Cypress.ResolvedConfigOptions, 'chromeWebSecurity' | 'supportFolder' | 'experimentalSourceRewriting' | 'fixturesFolder' | 'reporter' | 'reporterOptions' | 'screenshotsFolder' | 'pluginsFile' | 'supportFile' | 'integrationFolder' | 'baseUrl' | 'viewportHeight' | 'viewportWidth' | 'port' | 'experimentalInteractiveRunEvents'>>
 
 export interface Cfg extends ReceivedCypressOptions {
   projectRoot: string
@@ -675,7 +675,7 @@ export class ProjectBase<TServer extends ServerE2E | ServerCt> extends EE {
 
     if (theCfg.browsers) {
       theCfg.browsers = theCfg.browsers?.map((browser) => {
-        if (browser.family === 'chromium') {
+        if (browser.family === 'chromium' || theCfg.chromeWebSecurity) {
           return browser
         }
 

--- a/packages/server/test/unit/project_spec.js
+++ b/packages/server/test/unit/project_spec.js
@@ -255,6 +255,33 @@ This option will not have an effect in Some-other-name. Tests that rely on web s
         expect(cfg).ok
       })
     })
+
+    // https://github.com/cypress-io/cypress/issues/17614
+    it('only attaches warning to non-chrome browsers when chromeWebSecurity:true', async function () {
+      config.get.restore()
+      sinon.stub(config, 'get').returns({
+        integrationFolder,
+        browsers: [{ family: 'chromium', name: 'Canary' }, { family: 'some-other-family', name: 'some-other-name' }],
+        chromeWebSecurity: true,
+      })
+
+      await this.project.initializeConfig()
+      .then(() => {
+        const cfg = this.project.getConfig()
+
+        expect(cfg.chromeWebSecurity).eq(true)
+        expect(cfg.browsers).deep.eq([
+          {
+            family: 'chromium',
+            name: 'Canary',
+          },
+          {
+            family: 'some-other-family',
+            name: 'some-other-name',
+          },
+        ])
+      })
+    })
   })
 
   context('#initializeConfig', function () {

